sql/hive/src/test/java/org/apache/spark/sql/hive/JavaDataFrameSuite.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.hive;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.sql.*;
+import org.apache.spark.sql.hive.test.TestHive$;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.util.Utils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class JavaDataFrameSuite {
+    private transient JavaSparkContext sc;
+    private transient HiveContext hc;
+
+    DataFrame df;
+
+    private void checkAnswer(DataFrame actual, List<Row> expected) {
+        String errorMessage = QueryTest$.MODULE$.checkAnswer(actual, expected);
+        if (errorMessage != null) {
+            Assert.fail(errorMessage);
+        }
+    }
+
+    @Before
+    public void setUp() throws IOException {
+        hc = TestHive$.MODULE$;
+        sc = new JavaSparkContext(hc.sparkContext());
+
+        List<String> jsonObjects = new ArrayList<String>(10);
+        for (int i = 0; i < 10; i++) {
+            jsonObjects.add("{\"key\":" + i + ", \"value\":\"str" + i + "\"}");
+        }
+        df = hc.jsonRDD(sc.parallelize(jsonObjects));
+        df.registerTempTable("window_table");
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        // Clean up tables.
+        hc.sql("DROP TABLE IF EXISTS window_table");
+    }
+
+    @Test
+    public void saveTableAndQueryIt() {
+        checkAnswer(
+            df.select(
+                functions.avg("key").over(
+                    Window$.MODULE$.partitionBy("value")
```

Comment:
once you add WindowSpec class in object Window, change this Window$.MODULE$ thing into just Window